### PR TITLE
[ENG-6096] Fix incorrectly generated CAS login URL

### PR DESCRIPTION
**Pull request by cslzchen**
## Purpose

Fix the issue that CAS verification key login failed with `Invalid Verification Key Error` when user's username (i.e. primary email) contains `+`.

This bug was introduced by Python Upgrade where correct encoding for CAS login URL is decoded incorrectly before sending being set as a query parameter in the CAS logout URL, where `+` was decoded to ` `. Thus, the username CAS received is `test orcid@cos.io` instead of `test+orcid@cos.io`.

## Changes

* Remove `unquote_plus` when generating CAS login URLs, one for ORCiD SSO and the other for contributor claim.
* Updated unit tests

## QA Notes

* Test ORCiD SSO where user's primary email has `+` in it
* Test project contributor claim as an registered OSF user where the primary email has `+` in it

## Documentation

N/A

## Side Effects

N/A

## Ticket

https://openscience.atlassian.net/browse/ENG-6096

### CODE REVIEW

File: tests/test_auth.py [hunk 1]
```diff
@@ -135,10 +135,9 @@ def test_successful_external_login_cas_redirect(self, mock_service_validate, moc
         resp = cas.make_response_from_ticket(ticket, service_url)
         assert resp.status_code == 302, 'redirect to CAS login'
         assert quote_plus('/login?service=') in resp.location
-
-        # the valid username will be double quoted as it is furl quoted in both get_login_url and get_logout_url in order
-        assert quote_plus(f'username={user.username}') in resp.location
-        assert quote_plus(f'verification_key={user.verification_key}') in resp.location
+        # the valid username and verification key should be double-quoted
+        assert quote_plus(f'username={quote_plus(user.username)}') in resp.location
+        assert quote_plus(f'verification_key={quote_plus(user.verification_key)}') in resp.location
```
Comment (cslzchen) [inline tests/test_auth.py:140]: Note for CR: tests was incorrectly "FIXED" during python upgrade. This is now fixed.

**Review by mfraezz — APPROVED** [review]
LGTM ⭐

**Review by brianjgeiger — APPROVED** [review]
Seems good to me.

**Comment by cslzchen** [comment]
Rebase and merged during hotfix in https://github.com/CenterForOpenScience/osf.io/commit/c704152351b7d87940da3ef215bc411e537b6d73, consider this PR merged.

**Pull request closed without merging**